### ISSUE TITLE
[Question] Reconstructed images are gray

### ISSUE THREAD

**Issue by XinbeiCai** [KlingTeam/SVG-T2I#4]
**I am trying to run inference using the pre-trained svg_autoencoder model. I have successfully loaded the model with the corresponding config and checkpoint, but the reconstructed images appear gray, low-contrast, and washed out.**

<img width="666" height="333" alt="Image" src="https://github.com/user-attachments/assets/dcc0c4b1-1ca7-485f-8e7f-9eac1ddb5a43" />

code:
```import os
import sys
import torch
import numpy as np
from omegaconf import OmegaConf
from PIL import Image
from torchvision import transforms
from torchvision.utils import save_image
import torch.nn as nn


from ldm.util import instantiate_from_config

def load_model_from_config(config, ckpt, verbose=False):
    print(f"📂 Loading model from {ckpt}")
    pl_sd = torch.load(ckpt, map_location="cpu")
    if "state_dict" in pl_sd:
        sd = pl_sd["state_dict"]
    else:
        sd = pl_sd
        
    model = instantiate_from_config(config.model)

    m, u = model.load_state_dict(sd, strict=False)
    model.eval()
    return model

def process_image(image_path, size=256):
    """
    预处理：读取图片并归一化到 [-1, 1]
    """
    image = Image.open(img_path).convert("RGB").resize((TARGET_SIZE, TARGET_SIZE), Image.LANCZOS) # 丢弃 Alpha 通道，转为 RGB
    transform = transforms.Compose([
        transforms.Resize(size, interpolation=transforms.InterpolationMode.BICUBIC),
        transforms.CenterCrop(size),
        transforms.ToTensor(), # [0, 1]
        transforms.Normalize([0.5], [0.5]) #
    ])
    return transform(image).unsqueeze(0)


if __name__ == '__main__':
    # 1. 基础设置 
    device = "cuda" if torch.cuda.is_available() else "cpu"
    print(f"🚀 Running on device: {device}")

    # === 配置区域 (在这里修改路径) ===
    config_path = "configs/residual/autoencoder.yaml"
    ckpt_path = "pre-trained/autoencoder/svg_autoencoder-R-stage3.ckpt"
    img_path = "demo/dog.jpg"
    out_dir = "output"
    TARGET_SIZE = 256
    # ==============================

    # 2. 加载配置和模型
    config = OmegaConf.load(config_path)

    # 移除 Loss 配置 
    if "model" in config and "params" in config.model:
        if "lossconfig" in config.model.params:
            print("🔧 Removing Loss config for inference...")
            config.model.params.lossconfig.target = "torch.nn.Identity"
            config.model.params.lossconfig.params = {}

    model = load_model_from_config(config, ckpt_path)
    model = model.to(device)

    # 3. 数据预处理
    if not os.path.exists(img_path):
        raise FileNotFoundError(f"Input image not found: {img_path}")
        
    input_tensor = process_image(img_path, size=TARGET_SIZE).to(device)
    print(f"📥 Input Tensor Shape: {input_tensor.shape}")

    # 4. 运行模型 (Encode -> Decode)
    print("⚡ Running inference...")
    with torch.no_grad():
        # Encode
        z = model.encode(input_tensor)
        print(f"Encoded Latent Shape: {z.shape}")

        # Decode
        reconstructed = model.decode(z)
        print(f"Reconstructed Shape: {reconstructed.shape}")

        # === 调试信息 ===
        print(f"DEBUG: Recon Range [{reconstructed.min():.3f}, {reconstructed.max():.3f}]")

    # 5. 后处理与保存结果 
    
    # A. 对原始图进行反归一化
    orig_tensor = torch.clamp((input_tensor + 1.0) / 2.0, 0., 1.)
    reconstructed = torch.clamp((reconstructed + 1.0) / 2.0, 0., 1.)

    # 左右拼接 
    combined = torch.cat([orig_tensor, reconstructed], dim=3) 

    # 保存
    os.makedirs(out_dir, exist_ok=True)
    output_filename = os.path.join(out_dir, "comparison_result.png")
    save_image(combined, output_filename)
    
    print(f"🎉 Result saved to: {os.path.abspath(output_filename)}")
    print("   (Left: Original, Right: Reconstructed)")```




yaml:
```# Main training configuration - Using original target
training:
  logdir: "logs"
  scale_lr: false
  accumulate_grad_batches: 1
  save_every_n_train_steps: 20000
  save_every_n_epochs: 1

  trainer:
    devices: 8
    num_nodes: 1
    strategy: "ddp_find_unused_parameters_true"
    accelerator: "gpu"
    max_epochs: 1000
    precision: 32
  
  
# Model configuration
model:
  init_weight: null
  base_learning_rate: 1.0e-04
  target: ldm.models.dinov3_decoder_native_resolution.DinoDecoder
  params:    
    ckpt_path: null
    embed_dim: 32
    lossconfig:
      target: "ldm.modules.losses.LPIPSWithDiscriminatorDecoder"
      params:
        disc_start: 5000
        disc_weight: 0.5
    ddconfig:
      double_z: true
      z_channels: 392
      resolution: 256
      in_channels: 3
      out_ch: 3
      ch: 128
      ch_mult:
      - 1
      - 1
      - 2
      - 2
      - 4
      num_res_blocks: 2
      attn_resolutions: []
      dropout: 0.0

    dinoconfig:
      dinov3_location: dinov3
      model_name: dinov3_vits16plus
      weights: dinov3_vits16plus_pretrain_lvd1689m-4057cbaa.pth
    extra_vit_config: 
      output_dim: 8
      mask_ratio: 0.0
      use_outnorm: true
      vit_type: vit-rope-s

# Data configuration
data:
  target: utils.data_module_allinone.DataModuleFromConfigJson
  params:
    batch_size: 3 # batch size per GPU
    wrap: true
    train_resol: 256
    json_path: configs/example.jsonl```


Any help would be appreciated!

**Comment by XinbeiCai** [comment]
Issue resolved. I confirmed that this is an intentional asymmetric design.

Input: To align with pre-trained backbones (e.g., DINOv2), the input must be normalized using ImageNet mean/std statistics.

Output: The Decoder is trained to target the original image space (linear [-1, 1]). Therefore, the output does not require ImageNet denormalization; simple linear rescaling is sufficient.

**Comment by yaseryacoob** [comment]
@XinbeiCai  When I try ckpt_path = "pre-trained/autoencoder/svg_autoencoder-R-stage3.ckpt" with the respective .yaml I get an error (I also get the error when I try scripts/sample.sh with any R ckpt and yaml, but the P is alright. Can anybody clarify what's going on? It has to do with the residual 8 channels, but what is it?

```
size mismatch for decoder.conv_in.weight: copying a param with shape torch.Size([512, 392, 3, 3]) from checkpoint, the shape in current model is torch.Size([512, 384, 3, 3]).
'''

Thanks!
